Doc/c-api/time.rst

```diff
@@ -0,0 +1,132 @@
+range is around [-292.3 years; +292.3 years]. Using the Unix epoch (January
+1st, 1970), the supported date range is around [1677-09-21; 2262-04-11].
+
+Time formats:
+
+* Seconds.
+* Seconds as a floating pointer number (C :c:expr:`double`).
+* Milliseconds (10\ :sup:`-3` seconds).
+* Microseconds (10\ :sup:`-6` seconds).
+* 100 nanoseconds (10\ :sup:`-7` seconds), used on Windows.
+* Nanoseconds (10\ :sup:`-9` seconds).
+* :c:expr:`timeval` structure, 1 microsecond (10\ :sup:`-6` seconds).
+* :c:expr:`timespec` structure, 1 nanosecond (10\ :sup:`-9` seconds).
+
+Integer overflows are detected and raise :exc:`OverflowError`. Conversion to a
+resolution larger than 1 nanosecond is rounded correctly with the requested
+rounding mode. Available rounding modes:
+
+* Round towards minus infinity (-inf). For example, used to read a clock.
+* Round towards infinity (+inf). For example, used for timeout to wait "at
+  least" N seconds.
+* Round to nearest with ties going to nearest even integer. For example, used
+  to round from a Python float.
+* Round away from zero. For example, used for timeout.
+
+Some functions clamp the result in the range [PyTime_MIN; PyTime_MAX]. The
+caller doesn't have to handle errors and so doesn't need to hold the GIL to
+handle exceptions. For example, ``_PyTime_Add(t1, t2)`` computes ``t1+t2`` and
+clamps the result on overflow.
+
+Clocks:
+
+* System clock
+* Monotonic clock
+* Performance counter
+
+Internally, operations like ``(t * k / q)`` with integers are implemented in a
+way to reduce the risk of integer overflow. Such operation is used to convert a
+clock value expressed in ticks with a frequency to PyTime_t, like
+``QueryPerformanceCounter()`` with ``QueryPerformanceFrequency()`` on Windows.
+
+
+Types
+-----
+
+.. c:type:: PyTime_t
+
+   Timestamp type with subsecond precision: 64-bit signed integer.
+
+   This type can be used to store a duration. Indirectly, it can be used to
+   store a date relative to a reference date, such as the UNIX epoch.
+
+
+Constants
+---------
+
+.. c:var:: PyTime_t PyTime_MIN
+
+   Minimum value of the :c:type:`PyTime_t` type.
+   :c:macro`PyTime_MIN` nanoseconds is around -292.3 years.
+
+.. c:var:: PyTime_t PyTime_MAX
+
+   Maximum value of the :c:type:`PyTime_t` type.
+   :c:macro`PyTime_MAX` nanoseconds is around +292.3 years.
+
+
+Functions
+---------
+
+.. c:function:: double PyTime_AsSecondsDouble(PyTime_t t)
+
+   Convert a timestamp to a number of seconds as a C :c:expr:`double`.
+
+   The function cannot fail.
+
+
+.. c:function:: PyTime_t PyTime_GetMonotonicClock(void)
+
+   Get the time of a monotonic clock, i.e. a clock that cannot go backwards.
+   The clock is not affected by system clock updates. The reference point of
+   the returned value is undefined, so that only the difference between the
+   results of consecutive calls is valid.
+
+   If reading the clock fails, silently ignore the error and return 0.
```

Comment:
We're adding new functions here, so shouldn't we let them report errors? I don't think silent failures are what you'd want for a new API. Why not return `-1` (without exception) in the case of failures for all three get-time functions? Then users can decide themselves whether they want to handle or ignore the error. Or do you consider `0` a safe error indicator?